### PR TITLE
feat(e2e): introduce E2E framework

### PR DESCRIPTION
**Pull request by mhmxs**
## What this PR does / why we need it:

This change adds Chainsaw to drive E2E testing.

## Which issue(s) this PR resolves:

Resolves # https://github.com/linode/linode-cosi-driver/issues/52

### CODE REVIEW

File: test/Makefile [hunk 1]
```diff
@@ -25,6 +29,110 @@ clean: # Cleanup the test environment.
...
+	$(MAKE) clean-test-cluster
+
+_e2e-test:
+	$(CHAINSAW) test --parallel 1 --test-dir $(ROOT_DIR)/test/e2e --selector $(TEST_SELECTOR)
+
+test-cluster: ctlptl kind helm
+	$(CTLPTL) apply -f $(ROOT_DIR)/test/ctlptl-config.yaml
+	kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-api
+	kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-controller
+	$(HELM) install linode-cosi-driver \
```
Comment (mhmxs) [inline test/Makefile:45]: Do we need custom build or we are happy with the default image provided by Helm chart?

Comment (shanduur) [reply test/Makefile:45]: I would love to have there a custom image.

**Comment by shanduur** [comment]
Amazing work @mhmxs!

File: test/Makefile [hunk 2]
```diff
@@ -25,6 +29,110 @@ clean: # Cleanup the test environment.
...
+        $(ROOT_DIR)/helm/linode-cosi-driver \
+        --set=apiToken=$(LINODE_TOKEN) \
```
Comment (shanduur) [inline test/Makefile:47]:
```suggestion
        --set=apiToken=$(LINODE_TOKEN) \
        --set=driver.image.tag=$(VERSION) \
```

Also adding the following to the Makefile?

```
VERSION ?= $(shell git tag | tail -n 1 | grep '' || echo 'v0.0.0')$(shell git diff --quiet || git rev-parse HEAD | sed 's/\(.\{6\}\).*/-\1/')
```

Comment (shanduur) [reply test/Makefile:47]: Do you have an idea for loading the image into KinD's registry before installing the chart here?

Comment (mhmxs) [reply test/Makefile:47]: @shanduur im workong on it.

**Review by shanduur — APPROVED** [review]

**Pull request merged**